src/porepy/utils/array_operations.py

```diff
@@ -0,0 +1,262 @@
+"""The module contains various functions and classes useful for operations on numpy
+ndarrays.
+
+"""
+from __future__ import annotations
+
+import numpy as np
+from scipy.spatial import KDTree
+
+
+class SparseNdArray:
+    """Data structure for representing N-dimensional sparse arrays.
+
+    The underlying data format is a generalization of the COO-format (in the
+    scipy.sparse sense). The dimension is fixed upon initialization, but no bounds are
+    set on the maximum and minimum coordinates. The coordinates are restricted to
+    integers, thus, if the data to be stored lives a grid with non-integer 'grid-lines',
+    the data should be mapped prior to feeding it to the data. To do this, it is
+    recommended to consider the class AdaptiveInterpolationTable.
+
+    Args:
+        dims (int): Number of coordinate axes of the table.
+        value_dim: Dimension of the values to be represented in the array. Defaults to 1.
```

Comment:
What is the pattern we use for documentting attributes?

Reply:
As per @vlipovac's system, it should be done in init. I have updated the documentation style.

Reply:
Here is an example for documenting init args
https://github.com/pmgbergen/porepy/blob/ad_system/src/porepy/numerics/ad/ad_system.py#L183
Please use type annotations in the signature and omit the (int) and similar in the docstrings. Sphinx will parse the type annotation much better (and prettier). The type in the docstring is redundant.

And just below you will find attribute docs
https://github.com/pmgbergen/porepy/blob/ad_system/src/porepy/numerics/ad/ad_system.py#L209

Reply:
Thank you @vlipovac!